core/src/main/java/org/apache/accumulo/fate/Repo.java

```diff
@@ -34,16 +34,11 @@ default String getReturn() {
     return null;
   }
 
-  // return a string of the current status
-  default String getStatus() {
-    return "WARNING: Status has not been implemented yet.";
-  }
-
   // return the Goal of the current operation
   default Goal getGoal() {
-    return new Goal("WARNING: Status has not been implemented yet.");
+    return new Goal(0, 0, "WARNING: Operation has not implemented a goal.");
```

Comment:
In #1249 I was thinking the goal information could be provided when a fate transaction is seeded and stored in ZK.  It would not be obtained from each repo, instead status information could be related with each repo in the stack in ZK.  So to get the information for a fate transaction, get the goal for the fate transaction and then travese the stack getting the status information for each repo.

Reply:
For example maybe [this method](https://github.com/apache/accumulo/blob/7088cdec4fab5bb590645ab6dc10fb7b2021f50e/core/src/main/java/org/apache/accumulo/fate/Fate.java#L271) could be changed to :

```java
public void seedTransaction(long tid, Repo<T> repo, boolean autoCleanUp, String goalMessage)
```

Reply:
Thanks for the ideas. I hadn't thought of that.

Reply:
Also what do you think of trying to get the status reporting into 2.1 in a separate change? Seems like adding information to the serialized nodes shouldn't break whats already there.

Reply:
I think it would be a useful feature to have in 2.1 if we can get it done.  Thinking about it a bit more I am puzzling over how the status info would stored in ZK.  Was thinking it would go w/ the stack of repos, however in the case of undoing repos the status info would be lost.   Under normal circumstances repos are pushed onto a per fate tx stack in ZK as they are executed, so this stack could also have status info.  However when there is an error fate will start popping up repos off the stack and executing undo ops on them.  Because of this thinking it may make more sense to just have a per fate transaction list of status strings/objects in ZK.